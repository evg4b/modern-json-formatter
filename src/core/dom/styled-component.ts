import { registerStyles } from '@core/ui/helpers';

export abstract class StyledComponentElement extends HTMLElement {
  protected readonly shadow = this.attachShadow({ mode: 'closed' });
  protected readonly _styles: HTMLStyleElement;

  protected constructor(styles: string) {
    super();
    this._styles = registerStyles(this.shadow, styles);
  }
}
