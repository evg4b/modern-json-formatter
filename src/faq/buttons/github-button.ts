import { CustomElement } from '@core/dom';
import { BaseButtonElement } from './button';
import icon from './github-button-icon.svg';

@CustomElement('github-button')
export class GithubButton extends BaseButtonElement {
  constructor() {
    super(icon);
  }
}
