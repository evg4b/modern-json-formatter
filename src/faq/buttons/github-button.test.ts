import '@testing/svg.mock';
import { describe, expect, it } from '@jest/globals';
import { getShadowRoot } from '@testing/styled-component';
import { GithubButton } from './github-button';

describe('github-button', () => {
  it('should render button link', () => {
    const button = new GithubButton();
    button.attributeChangedCallback('href', '', 'https://githib.com');
    button.attributeChangedCallback('title', '', 'Github Repo');
    const root = getShadowRoot(button);
    expect(root.innerHTML).toMatchSnapshot();
  });
});
