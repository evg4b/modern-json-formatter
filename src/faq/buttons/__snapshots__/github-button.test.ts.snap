// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`github-button should render button link 1`] = `"<style type="text/css" rel="stylesheet" role="presentation"></style><a target="_blank" href="https://githib.com" title="Github Repo"><svg>github-button-icon</svg></a>"`;
